### PR TITLE
Fix #3248 & #3251 - Update metrics and column profile

### LINKED ISSUE #3248: Data Quality - Review unique vs distinct

### PR DESCRIPTION
**Pull request by pmbrull**
### Describe your changes :

This PR fixes https://github.com/open-metadata/OpenMetadata/issues/3248 and fixes https://github.com/open-metadata/OpenMetadata/issues/3251.

We have:
- Removed `missingPercentage` and `missingCount` from the columnProfile. It just makes sense in tests.
- Added `distinctProportion` to columnProfile
- Refactored and properly differentiated unique and distinct metrics
- Updated metrics test with more data
- Updated profiler QueryMetric runner
- Added missing `@property` to a bunch of metrics.

Thanks

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have performed a self-review of my own. 
- [ ] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: ingestion/src/metadata/orm_profiler/metrics/composed/distinct_ratio.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    def metric_type(self):
+        """
+        Override default metric_type definition as
+        we now don't care about the column
+        """
+        return float
+
+    def fn(self, res: Dict[str, Any]) -> Optional[float]:
+        """
+        Safely compute null ratio based on the profiler
```
Comment (Sachin-chaurasiya) [inline ingestion/src/metadata/orm_profiler/metrics/composed/distinct_ratio.py:46]:
```suggestion
        Safely compute distinct ratio based on the profiler
```

File: openmetadata-ui/pom.xml [hunk 2]
```diff
@@ -95,7 +95,7 @@
       <plugin>
         <groupId>com.github.eirslett</groupId>
         <artifactId>frontend-maven-plugin</artifactId>
-        <version>1.11.0</version>
+        <version>1.12.1</version>
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/pom.xml:98]: @pmbrull , can we revert these file changes. because it might break UI build.

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]
Few comments

File: openmetadata-ui/pom.xml (hunk 2 above)
Comment (pmbrull) [reply openmetadata-ui/pom.xml:98]: ups didn't mean to push this, good catch. I just need it in local to make it run. Reverting

**Comment by pmbrull** [comment]
> Few comments

Thanks, @Sachin-chaurasiya. Pushed the changes 🙏

**Review by Sachin-chaurasiya — APPROVED** [review]

**Pull request merged**